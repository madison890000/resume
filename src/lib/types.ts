export enum Degree {
    Bachelor = 'Bachelor',
    Master = 'Master',
    PhD = 'PhD',
}

export type Gender = 'male' | 'female';


export type Achievement = {
    text: string;
    categories: string[];
};
export type SolutionsOfHowToImplement = string;

export enum JobPositionLevel {
    Staff = 'Staff',
    Senior = 'Senior',
    Middle = '',
    Junior = 'Junior',
}

export type JobPosition =
    'Software Engineer'
    | 'Front-End Engineer'
    | 'Back-End Engineer'
    | 'Full-Stack Engineer';

export type JobResponsibility = string;
export type FEUILib = 'Material UI' | 'antd';
export type ReactFramework = 'NextJs' | 'Create React App' | 'umi';
export type Framework = 'React' | 'Vue' | 'Angular' | 'jQuery';
export type CodeLanguage = 'JAVA' | 'C#' | 'JavaScript';
export type OtherSkillLabels = 'NodeJs' | 'HTML5' | 'CSS3' | 'Webpack';
export type WayOfDevelopment = 'Agile' | 'Waterfall';

export type Keyword =
    Framework
    & CodeLanguage
    & OtherSkillLabels
    & FEUILib
    & ReactFramework
    & WayOfDevelopment;

export enum SkillLevel {
    understand = 'understand',
    familiar = 'familiar',
    proficient = 'proficient',
}

export enum Importance {
    important,
    normal,
    less,
}
export enum CompanyType {
    Enterprise,
    Startup,
}
