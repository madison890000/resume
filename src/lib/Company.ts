import { CompanyType } from './types';

interface ICompany {
    name: string;
    location?: string;
    type?: CompanyType;
}

export default class Company {
    public name: string;
    public type?: CompanyType;
    public location?: string;

    constructor({ name, location, type }: ICompany) {
        this.name = name;
        this.location = location;
        this.type = type;
    }

}
