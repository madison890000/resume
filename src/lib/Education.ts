import { Degree } from './types';


interface IEducation {
    college: string;
    major: string;
    degree: Degree;
    start: Date;
    end: Date;
}

export default class Education {
    public college: string;
    public major: string;
    public degree: Degree;
    public start: Date;
    public end: Date;

    constructor({ college,major, degree, start, end }: IEducation) {
        this.major = major;
        this.college = college;
        this.degree = degree;
        this.start = start;
        this.end = end;
    }
}
