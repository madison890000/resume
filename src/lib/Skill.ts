import { Importance, SkillLevel } from './types';


export default class Skill {
    public name: string;
    public level: SkillLevel;
    public importance: Importance;
    public ages: number;

    constructor(name: string, level: SkillLevel, ages: number, importance: Importance) {
        this.name = name;
        this.level = level;
        this.ages = ages;
        this.importance = importance;
    }

}
