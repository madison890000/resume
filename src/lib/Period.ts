import Company from './Company';
import { v4 } from 'uuid';
import {
    Achievement,
    JobPosition,
    JobPositionLevel,
    Keyword,
    SolutionsOfHowToImplement
} from './types';

interface IPeriod {
    start: Date;
    end: Date;
    keywords: Keyword[];
    company: Company;
    jobPositionLevel: JobPositionLevel;
    jobPosition: JobPosition;
    achievements: Achievement[];
    jobSummaries: string[];
    solutionsOfHowToImplement: SolutionsOfHowToImplement[];
}

export default class Period {
    readonly id: string;
    public start: Date;
    public end: Date;
    public keywords: Keyword[];
    public achievements: Achievement[];
    public jobSummaries: string[];
    public company: Company;
    public jobPositionLevel: JobPositionLevel;
    public jobPosition: JobPosition;
    public solutionsOfHowToImplement: SolutionsOfHowToImplement[];

    constructor({solutionsOfHowToImplement, company, jobSummaries, jobPosition, jobPositionLevel, achievements, keywords, start, end }: IPeriod) {
        this.id = v4();
        this.company = company;
        this.start = start;
        this.end = end;
        this.achievements = achievements;
        this.jobSummaries = jobSummaries;
        this.keywords = keywords;
        this.jobPositionLevel = jobPositionLevel;
        this.jobPosition = jobPosition;
        this.solutionsOfHowToImplement = solutionsOfHowToImplement;
    }

}
