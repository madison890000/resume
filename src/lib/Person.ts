import Education from './Education';
import Period from './Period';
import { Gender } from './types';
import Skill from './Skill';


interface IPerson {
    firstName: string;
    lastName: string;
    email: string;
    birthDay: Date;
    gender: Gender;
    cellphone: string;
    country: string;
    description: string[];
    educations?: Education[];
    periods?: Period[];
}


export default class Person {
    public firstName: string;
    public lastName: string;
    public email: string;
    public gender: Gender;
    public cellphone: string;
    public country: string;
    public birthDay: Date;
    public educations: Education[];
    public periods: Period[];
    public skills: Skill[];
    public description: string[];

    constructor({ firstName,description, birthDay, lastName, email, cellphone, country, gender }: IPerson) {
        this.educations = [];
        this.periods = [];
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.cellphone = cellphone;
        this.birthDay = birthDay;
        this.country = country;
        this.gender = gender;
        this.skills = [];
        this.description = description;
    }

    addEducations(educations: Education[]) {
        this.educations = this.educations.concat(educations);
    }

    addPeriods(periods: Period[]) {
        this.periods = this.periods.concat(periods);
    }

    addSkills(skills: Skill[]) {
        this.skills = this.skills.concat(skills);
    }

    get reversedPeriods() {
        return [...this.periods].reverse()
    }

}
