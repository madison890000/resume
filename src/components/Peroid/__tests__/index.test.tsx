import React from 'react';
import Period from '../index';
import renderer from 'react-test-renderer';
import { Achievement, JobPositionLevel, Keyword } from '../../../lib/types';

test('renders Period', () => {
    const testCompanyName = 'test company name'
    const start = new Date('2012-01-01');
    const end = new Date('2013-01-01');
    const keywords = ['React'] as Keyword[];
    const jobPositionLevel = JobPositionLevel.Senior;
    const jobPosition = 'Software Engineer';
    const achievements = [] as Achievement[];
    const jobSummaries = [] as string[];
    const PeriodDom = renderer.create(
        <Period
            start={start}
            end={end}
            keywords={keywords}
            jobPositionLevel={jobPositionLevel}
            jobPosition={jobPosition}
            achievements={achievements}
            jobSummaries={jobSummaries}
            companyName={testCompanyName}
        />
    )
        .toJSON();
    expect(PeriodDom)
        .toMatchSnapshot();
});
