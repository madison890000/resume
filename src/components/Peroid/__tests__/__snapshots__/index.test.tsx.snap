// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Period 1`] = `
Array [
  <div
    className="period"
  >
    <div>
      <div
        className="header"
      >
        <span
          className="job"
        >
          <span>
            Senior
          </span>
          <span>
            Software Engineer
          </span>
        </span>
        <span
          className="companyName"
        >
          <span>
            test company name
            <span />
          </span>
        </span>
        <span>
          <span
            className="filled"
          >
            React
          </span>
        </span>
      </div>
    </div>
    <div
      className="times"
    >
      <span>
        Jan, 2012
      </span>
      <span>
        -
      </span>
      <span>
        Jan, 2013
      </span>
    </div>
  </div>,
  <div>
    <h5>
      Job summaries:
    </h5>
    <ul />
  </div>,
  <div>
    <h5>
      Achievements:
    </h5>
    <ul
      className="achievements"
    />
  </div>,
]
`;
